module-apps/application-meditation/data/OptionsData.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "OptionsData.hpp"
+#include <log/log.hpp>
+
+namespace gui
+{
+    OptionsData::OptionsData(OptionParams params) : params(params)
+    {}
+
+    [[nodiscard]] std::chrono::seconds OptionsData::getPreparationTime() const noexcept
+    {
+        return params.preparationTime.get();
+    }
+
+    void OptionsData::setPreparationTime(std::chrono::seconds value)
+    {
+        const auto it = std::find(std::begin(preparationTimes), std::end(preparationTimes), value);
+        if (it == std::end(preparationTimes)) {
+            LOG_ERROR("Incorrect parameter value! The default value is set.");
```

Comment:
How about returning error code instead of printing log? This way you do not need to include log dependency to this module. Error condition then would be handled by the caller(you can simply insert logging there)

Reply:
done